src/url.rs

```diff
@@ -177,15 +177,25 @@ impl PyUrl {
         };
         let mut url = format!("{scheme}://{url_host}");
         if let Some(path) = path {
-            url.push('/');
-            url.push_str(path);
+            if !url.ends_with('/') {
+                url.push('/');
+            }
```

Comment:
I feel like we should expect that `url` _never_ ends with `/`, as that's technically not a valid hostname? I wonder if there's a case to raise an error if `url_host` contains `/`?